app/src/main/java/com/example/tasky/feature_agenda/presentation/AgendaViewModel.kt

```diff
@@ -165,4 +172,22 @@ class AgendaViewModel @Inject constructor(
             date.toFormatted_MM_DD_YYYY()
         }
     }
+
+    fun fabActionClicked(action: Action) {
+        viewModelScope.launch {
+            when (action) {
+                Action.Event -> {
+                    _viewEvent.send(AgendaViewEvent.NavigateToEventScreen)
+                }
+
+                Action.Task -> {
+                    _viewEvent.send(AgendaViewEvent.NavigateToTaskScreen)
+                }
+
+                Action.Reminder -> {
+                    _viewEvent.send(AgendaViewEvent.NavigateToReminderScreen)
+                }
+            }
+        }
+    }
```

Comment:
You already have access to the nav controller in the UI, why take this detour over the ViewModel just to send an event back?